python/perf-kernels/streamk/03-matrix-multiplication-stream-k.py

```diff
@@ -0,0 +1,216 @@
+
+    _debug = True
+
+    @staticmethod
+    def set_debug(debug: bool):
+        matmul._debug = debug
+
+    @staticmethod
+    def _call(a: torch.Tensor, b: torch.Tensor, c: torch.Tensor, bias: torch.Tensor, P: torch.Tensor,
+              locks: torch.Tensor, total_programs_streamk: int, BLK_M: int, BLK_N: int, BLK_K: int, gsize_m: int,
+              two_tiles: bool, num_stages: int, num_warps: int, waves_per_eu: int, mfmaInstrSize: int, kpack: int):
+
+        #        assert a.is_contiguous() and b.is_contiguous(), "non-contiguous inputs are not supported"
+        # checks constraints
+        assert a.shape[1] == b.shape[0], "incompatible dimensions"
+        M, K = a.shape
+        _, N = b.shape
+
+        total_blocks_M = triton.cdiv(M, BLK_M)
+        total_blocks_N = triton.cdiv(N, BLK_N)
+        iters_per_tile = triton.cdiv(K, BLK_K)
+        total_tiles = total_blocks_M * total_blocks_N
+        even_k = K % BLK_K == 0
+
+        if total_programs_streamk > 0:  # Stream-K
+            # last wave may occupy less than total_programs_streamk SMs
+            total_tiles_streamk = total_tiles % total_programs_streamk
+            # for two-tile Stream-K + data-parallel from original paper
+            #            if two_tiles and total_tiles - total_tiles_streamk > total_programs_streamk:
+            #                total_tiles_streamk += total_programs_streamk
+            # remaining tiles are computed using classical blocking
+            total_blocking_tiles = total_tiles - total_tiles_streamk
+            total_iters_streamk = total_tiles_streamk * iters_per_tile
+            # iterations related to full waves
+            total_full_tiles_streamk = total_iters_streamk // total_programs_streamk
+            # iterations related to last (partial) wave
+            total_partial_tiles_streamk = total_iters_streamk % total_programs_streamk
+
+        else:  # all tiles are computed using classical blocking
+            total_blocking_tiles = total_tiles
+            total_tiles_streamk = 0
+            total_full_tiles_streamk = 0
+            total_partial_tiles_streamk = 0
+            total_iters_streamk = 0
+
+        if matmul._debug:
+            print(f"M,N,K={M},{N},{K} ; BLK_M,N,K={BLK_M},{BLK_N},{BLK_K}")
+            print(f"{total_blocks_M=} x {total_blocks_N=} = {total_tiles=}")
+            print(f"{total_tiles_streamk=} + {total_blocking_tiles=} = {total_tiles=}")
+            print(f"{total_programs_streamk=}")
+            print(f"{total_blocking_tiles=}")
+            print(f"{total_full_tiles_streamk=}")
+            print(f"{iters_per_tile=}")
+            print(f"{total_iters_streamk=}")
+            print("total_remainder_iters_streamk=", total_partial_tiles_streamk)
+        use_bias = False
+        # compute grid (work to do per SM on the first wave)
+        grids = total_programs_streamk
+        stride_bias = bias.stride(0) if use_bias else 0
+        #  P=P*0.0
+        #  locks=locks*0
```

Comment:
We never need to reset locks, comment can be removed.

Reply:
removed